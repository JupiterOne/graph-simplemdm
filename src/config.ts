import {
  IntegrationExecutionContext,
  IntegrationValidationError,
  IntegrationInstanceConfigFieldMap,
  IntegrationInstanceConfig,
} from '@jupiterone/integration-sdk-core';
import { createAPIClient } from './client';

export const instanceConfigFields: IntegrationInstanceConfigFieldMap = {
  apiKey: {
    type: 'string',
    mask: true,
  },
};

export interface IntegrationConfig extends IntegrationInstanceConfig {
  apiKey: string;
}

export async function validateInvocation(
  context: IntegrationExecutionContext<IntegrationConfig>,
) {
  const { config } = context.instance;

  if (!config.apiKey) {
    throw new IntegrationValidationError('Config requires all of {apiKey}');
  }

  const apiClient = createAPIClient(config);
  await apiClient.verifyAuthentication();
}
