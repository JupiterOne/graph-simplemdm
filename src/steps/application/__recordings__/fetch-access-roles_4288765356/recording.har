{
  "log": {
    "_recordingName": "fetch-access-roles",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "6.0.5"
    },
    "entries": [
      {
        "_id": "265fa08212f502e8cac45129d0af734c",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "a.simplemdm.com"
            }
          ],
          "headersSize": 320,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://a.simplemdm.com/api/v1/account"
        },
        "response": {
          "bodySize": 106,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 106,
            "text": "{\"data\":{\"type\":\"account\",\"attributes\":{\"name\":\"JupiterOne, Inc.\",\"apple_store_country_code\":\"US\"}}}"
          },
          "cookies": [
            {
              "expires": "2022-06-24T01:18:17.000Z",
              "httpOnly": true,
              "name": "_mdm_session",
              "path": "/",
              "sameSite": "Lax",
              "secure": true,
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "date",
              "value": "Thu, 23 Jun 2022 15:18:17 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "content-length",
              "value": "106"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "server",
              "value": "Apache/2.4.41 (Ubuntu)"
            },
            {
              "name": "cache-control",
              "value": "max-age=0, private, must-revalidate"
            },
            {
              "name": "vary",
              "value": "Accept,Accept-Encoding"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains"
            },
            {
              "name": "referrer-policy",
              "value": "strict-origin-when-cross-origin"
            },
            {
              "name": "x-permitted-cross-domain-policies",
              "value": "none"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block"
            },
            {
              "name": "x-request-id",
              "value": "9ec697b8-20b6-4885-bc7b-819a991bd43d"
            },
            {
              "name": "x-download-options",
              "value": "noopen"
            },
            {
              "name": "x-frame-options",
              "value": "sameorigin"
            },
            {
              "name": "x-runtime",
              "value": "0.019824"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "content-security-policy",
              "value": "frame-ancestors 'self'"
            },
            {
              "name": "x-powered-by",
              "value": "Phusion Passenger(R) 6.0.13"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "etag",
              "value": "W/\"d5169a64d6e846c2d95f0c0945091f42-gzip\""
            },
            {
              "name": "status",
              "value": "200 OK"
            }
          ],
          "headersSize": 1186,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-06-23T15:18:16.626Z",
        "time": 806,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 806
        }
      },
      {
        "_id": "6464f1a0c9c41ef33ebd77f45a3ded39",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "a.simplemdm.com"
            }
          ],
          "headersSize": 327,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "limit",
              "value": "100"
            }
          ],
          "url": "https://a.simplemdm.com/api/v1/apps?limit=100"
        },
        "response": {
          "bodySize": 173,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 173,
            "text": "{\"data\":[{\"type\":\"app\",\"id\":490899,\"attributes\":{\"name\":\"SimpleMDM\",\"app_type\":\"apple store\",\"itunes_store_id\":847612226,\"bundle_identifier\":\"com.unwiredrev.DeviceLink.public\"}}],\"has_more\":false}"
          },
          "cookies": [
            {
              "expires": "2022-06-24T02:29:26.000Z",
              "httpOnly": true,
              "name": "_mdm_session",
              "path": "/",
              "sameSite": "Lax",
              "secure": true,
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "date",
              "value": "Thu, 23 Jun 2022 16:29:26 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "content-length",
              "value": "173"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "server",
              "value": "Apache/2.4.41 (Ubuntu)"
            },
            {
              "name": "cache-control",
              "value": "max-age=0, private, must-revalidate"
            },
            {
              "name": "vary",
              "value": "Accept,Accept-Encoding"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains"
            },
            {
              "name": "referrer-policy",
              "value": "strict-origin-when-cross-origin"
            },
            {
              "name": "x-permitted-cross-domain-policies",
              "value": "none"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block"
            },
            {
              "name": "x-request-id",
              "value": "a0e463d8-bdc0-47b1-8045-716dac53cd79"
            },
            {
              "name": "x-download-options",
              "value": "noopen"
            },
            {
              "name": "x-frame-options",
              "value": "sameorigin"
            },
            {
              "name": "x-runtime",
              "value": "0.019264"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "content-security-policy",
              "value": "frame-ancestors 'self'"
            },
            {
              "name": "x-powered-by",
              "value": "Phusion Passenger(R) 6.0.13"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "etag",
              "value": "W/\"e26506b90e2513b1ebdbdfc8feb21cff-gzip\""
            },
            {
              "name": "status",
              "value": "200 OK"
            }
          ],
          "headersSize": 1192,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-06-23T16:29:25.791Z",
        "time": 858,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 858
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
