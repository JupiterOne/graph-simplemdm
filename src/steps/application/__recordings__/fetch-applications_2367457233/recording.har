{
  "log": {
    "_recordingName": "fetch-applications",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "6.0.5"
    },
    "entries": [
      {
        "_id": "265fa08212f502e8cac45129d0af734c",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "a.simplemdm.com"
            }
          ],
          "headersSize": 320,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://a.simplemdm.com/api/v1/account"
        },
        "response": {
          "bodySize": 106,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 106,
            "text": "{\"data\":{\"type\":\"account\",\"attributes\":{\"name\":\"JupiterOne, Inc.\",\"apple_store_country_code\":\"US\"}}}"
          },
          "cookies": [
            {
              "expires": "2022-07-16T01:23:46.000Z",
              "httpOnly": true,
              "name": "_mdm_session",
              "path": "/",
              "sameSite": "Lax",
              "secure": true,
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "date",
              "value": "Fri, 15 Jul 2022 15:23:46 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "content-length",
              "value": "106"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "server",
              "value": "Apache/2.4.41 (Ubuntu)"
            },
            {
              "name": "cache-control",
              "value": "max-age=0, private, must-revalidate"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains"
            },
            {
              "name": "referrer-policy",
              "value": "strict-origin-when-cross-origin"
            },
            {
              "name": "x-permitted-cross-domain-policies",
              "value": "none"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-request-id",
              "value": "4fe19437-4d0b-43da-975a-4d315a43755a"
            },
            {
              "name": "x-download-options",
              "value": "noopen"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-runtime",
              "value": "0.015835"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "content-security-policy",
              "value": "frame-ancestors 'self'"
            },
            {
              "name": "x-powered-by",
              "value": "Phusion Passenger(R) 6.0.14"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "etag",
              "value": "W/\"d5169a64d6e846c2d95f0c0945091f42-gzip\""
            },
            {
              "name": "status",
              "value": "200 OK"
            },
            {
              "name": "vary",
              "value": "Accept-Encoding"
            }
          ],
          "headersSize": 1165,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-07-15T15:23:45.767Z",
        "time": 827,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 827
        }
      },
      {
        "_id": "5d1b17a280074bb4f53cdbabad009d9e",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "a.simplemdm.com"
            }
          ],
          "headersSize": 330,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "limit",
              "value": "100"
            }
          ],
          "url": "https://a.simplemdm.com/api/v1/devices?limit=100"
        },
        "response": {
          "bodySize": 1201,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 1201,
            "text": "{\"data\":[{\"type\":\"device\",\"id\":897636,\"attributes\":{\"name\":\"iPhone 6S - rose gold\",\"last_seen_at\":\"2022-07-15T15:10:05.000Z\",\"last_seen_ip\":\"136.36.62.44\",\"status\":\"enrolled\",\"enrollment_channels\":[\"device\"],\"device_name\":\"Samuelâ€™s iPhone\",\"os_version\":\"13.3\",\"build_version\":\"17C54\",\"model_name\":\"iPhone 6s\",\"model\":\"MKQD2LL\",\"product_name\":\"iPhone8,1\",\"unique_identifier\":\"d483069fe5f5b6114f01abb49d291125e399d5de\",\"serial_number\":\"FK1QQ3ATGRY0\",\"processor_architecture\":null,\"imei\":\"35 325807 416175 0\",\"meid\":\"35325807416175\",\"device_capacity\":48.6917,\"available_device_capacity\":41.396,\"battery_level\":\"100%\",\"modem_firmware_version\":\"7.30.02\",\"iccid\":\"8901 4104 2785 9535 3006\",\"bluetooth_mac\":\"6c:72:e7:d2:61:f8\",\"ethernet_macs\":[],\"wifi_mac\":\"6c:72:e7:d2:61:f7\",\"current_carrier_network\":null,\"sim_carrier_network\":null,\"subscriber_carrier_network\":\"AT&T\",\"carrier_settings_version\":\"40.0\",\"phone_number\":\"+18013684495\",\"voice_roaming_enabled\":false,\"data_roaming_enabled\":false,\"is_roaming\":false,\"subscriber_mcc\":\"310\",\"subscriber_mnc\":\"410\",\"simmnc\":null,\"current_mcc\":\"310\",\"current_mnc\":\"260\",\"hardware_encryption_caps\":3,\"passcode_present\":true,\"passcode_compliant\":true,\"passcode_compliant_with_profiles\":true,\"is_supervised\":false,\"is_dep_enrollment\":false,\"is_user_approved_enrollment\":null,\"is_device_locator_service_enabled\":false,\"is_do_not_disturb_in_effect\":false,\"personal_hotspot_enabled\":false,\"itunes_store_account_is_active\":false,\"cellular_technology\":3,\"last_cloud_backup_date\":null,\"is_activation_lock_enabled\":false,\"is_cloud_backup_enabled\":false,\"filevault_enabled\":false,\"filevault_recovery_key\":null,\"lost_mode_enabled\":false,\"firmware_password_enabled\":false,\"recovery_lock_password_enabled\":false,\"remote_desktop_enabled\":null,\"firmware_password\":null,\"recovery_lock_password\":null,\"firewall\":{\"enabled\":null,\"block_all_incoming\":null,\"stealth_mode\":null},\"system_integrity_protection_enabled\":null,\"os_update\":{\"automatic_os_installation_enabled\":null,\"automatic_app_installation_enabled\":null,\"automatic_check_enabled\":null,\"automatic_security_updates_enabled\":null,\"background_download_enabled\":null,\"catalog_url\":null,\"default_catalog\":null,\"perform_periodic_check\":null,\"previous_scan_date\":null,\"previous_scan_result\":null},\"service_subscriptions\":[{\"slot\":\"CTSubscriptionSlotOne\",\"carrier_settings_version\":\"40.0\",\"current_carrier_network\":\"\",\"current_mcc\":\"310\",\"current_mnc\":\"260\",\"iccid\":\"8901 4104 2785 9535 3006\",\"imei\":\"35 325807 416175 0\",\"is_data_preferred\":true,\"is_voice_preferred\":true,\"label\":\"USER_LABEL_PRIMARY\",\"meid\":\"35325807416175\"}],\"location_latitude\":null,\"location_longitude\":null,\"location_accuracy\":null,\"location_updated_at\":null},\"relationships\":{\"device_group\":{\"data\":{\"type\":\"device_group\",\"id\":106751}}}}],\"has_more\":false}"
          },
          "cookies": [
            {
              "expires": "2022-07-16T01:23:47.000Z",
              "httpOnly": true,
              "name": "_mdm_session",
              "path": "/",
              "sameSite": "Lax",
              "secure": true,
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "date",
              "value": "Fri, 15 Jul 2022 15:23:47 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "content-length",
              "value": "1201"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "server",
              "value": "Apache/2.4.41 (Ubuntu)"
            },
            {
              "name": "cache-control",
              "value": "max-age=0, private, must-revalidate"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains"
            },
            {
              "name": "referrer-policy",
              "value": "strict-origin-when-cross-origin"
            },
            {
              "name": "x-permitted-cross-domain-policies",
              "value": "none"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-request-id",
              "value": "8227f88c-b620-44a3-8d7e-da9a9666f63f"
            },
            {
              "name": "x-download-options",
              "value": "noopen"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-runtime",
              "value": "0.023202"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "content-security-policy",
              "value": "frame-ancestors 'self'"
            },
            {
              "name": "x-powered-by",
              "value": "Phusion Passenger(R) 6.0.14"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "etag",
              "value": "W/\"620074eb059428bd687a84272363b10c-gzip\""
            },
            {
              "name": "status",
              "value": "200 OK"
            },
            {
              "name": "vary",
              "value": "Accept-Encoding"
            }
          ],
          "headersSize": 1164,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-07-15T15:23:46.639Z",
        "time": 802,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 802
        }
      },
      {
        "_id": "5b1973bf3478c84cec898e138c4e770b",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "a.simplemdm.com"
            }
          ],
          "headersSize": 342,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://a.simplemdm.com/api/v1/devices/897636/installed_apps"
        },
        "response": {
          "bodySize": 256,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 256,
            "text": "{\"data\":[{\"type\":\"installed_app\",\"id\":6819243195,\"attributes\":{\"name\":\"SimpleMDM\",\"identifier\":\"com.unwiredrev.DeviceLink.public\",\"version\":\"2.1.2.0\",\"short_version\":\"2.1.2\",\"bundle_size\":19329024,\"dynamic_size\":90112,\"managed\":true,\"discovered_at\":\"2022-07-14T16:10:37.000Z\",\"last_seen_at\":\"2022-07-15T15:10:05.000Z\"}}],\"has_more\":false}"
          },
          "cookies": [
            {
              "expires": "2022-07-16T01:23:47.000Z",
              "httpOnly": true,
              "name": "_mdm_session",
              "path": "/",
              "sameSite": "Lax",
              "secure": true,
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "date",
              "value": "Fri, 15 Jul 2022 15:23:48 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "content-length",
              "value": "256"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "server",
              "value": "Apache/2.4.41 (Ubuntu)"
            },
            {
              "name": "cache-control",
              "value": "max-age=0, private, must-revalidate"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains"
            },
            {
              "name": "referrer-policy",
              "value": "strict-origin-when-cross-origin"
            },
            {
              "name": "x-permitted-cross-domain-policies",
              "value": "none"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-request-id",
              "value": "714d92a9-5309-489d-9599-18340f24b27b"
            },
            {
              "name": "x-download-options",
              "value": "noopen"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-runtime",
              "value": "0.020139"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "content-security-policy",
              "value": "frame-ancestors 'self'"
            },
            {
              "name": "x-powered-by",
              "value": "Phusion Passenger(R) 6.0.14"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "etag",
              "value": "W/\"ef718095bd21c2b2c51f30392b6c8666-gzip\""
            },
            {
              "name": "status",
              "value": "200 OK"
            },
            {
              "name": "vary",
              "value": "Accept-Encoding"
            }
          ],
          "headersSize": 1169,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-07-15T15:23:47.466Z",
        "time": 612,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 612
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
