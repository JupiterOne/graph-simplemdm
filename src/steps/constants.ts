import {
  RelationshipClass,
  StepEntityMetadata,
  StepRelationshipMetadata,
} from '@jupiterone/integration-sdk-core';

export const Steps = {
  ACCOUNT: 'fetch-account',
  APPLICATIONS: 'fetch-applications',
  DEVICES: 'fetch-devices',
  USERS: 'fetch-users',
};

export const Entities: Record<
  'ACCOUNT' | 'APPLICATION' | 'DEVICE' | 'USER',
  StepEntityMetadata
> = {
  ACCOUNT: {
    resourceName: 'Account',
    _type: 'simplemdm_account',
    _class: ['Account'],
  },
  APPLICATION: {
    resourceName: 'Application',
    _type: 'simplemdm_application',
    _class: ['Application'],
  },
  DEVICE: {
    resourceName: 'Device',
    _type: 'simplemdm_device',
    _class: ['Device'],
  },
  USER: {
    resourceName: 'User',
    _type: 'simplemdm_user',
    _class: ['User'],
  },
};

export const Relationships: Record<
  'ACCOUNT_HAS_APPLICATION' | 'ACCOUNT_HAS_DEVICE' | 'DEVICE_HAS_USER',
  StepRelationshipMetadata
> = {
  ACCOUNT_HAS_APPLICATION: {
    _type: 'simplemdm_account_has_application',
    sourceType: Entities.ACCOUNT._type,
    _class: RelationshipClass.HAS,
    targetType: Entities.APPLICATION._type,
  },
  ACCOUNT_HAS_DEVICE: {
    _type: 'simplemdm_account_has_device',
    sourceType: Entities.ACCOUNT._type,
    _class: RelationshipClass.HAS,
    targetType: Entities.DEVICE._type,
  },
  DEVICE_HAS_USER: {
    _type: 'simplemdm_device_has_user',
    sourceType: Entities.DEVICE._type,
    _class: RelationshipClass.HAS,
    targetType: Entities.USER._type,
  },
};
