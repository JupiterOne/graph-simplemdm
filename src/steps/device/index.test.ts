import { executeStepWithDependencies } from '@jupiterone/integration-sdk-testing';

import { buildStepTestConfigForStep } from '../../../test/config';
import { Recording, setupProjectRecording } from '../../../test/recording';
import { Steps } from '../constants';
import { omit } from 'lodash';

// See test/README.md for details
let recording: Recording;
afterEach(async () => {
  await recording.stop();
});

test('fetch-devices', async () => {
  recording = setupProjectRecording({
    directory: __dirname,
    name: 'fetch-devices',
  });

  const stepConfig = buildStepTestConfigForStep(Steps.DEVICES);
  const stepResult = await executeStepWithDependencies(stepConfig);
  expect(stepResult).toMatchStepMetadata(stepConfig);
  expect(
    stepResult.collectedEntities.map((e) => omit(e, ['_rawData'])),
  ).toMatchSnapshot();
});
