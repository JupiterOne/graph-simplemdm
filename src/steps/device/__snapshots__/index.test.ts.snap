// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetch-devices 1`] = `
Array [
  Object {
    "_class": Array [
      "Device",
    ],
    "_key": "simplemdm_device:872292",
    "_type": "simplemdm_device",
    "bluetoothMac": "14:7D:DA:4E:50:6D",
    "buildVersion": "21A559",
    "category": "other",
    "createdOn": undefined,
    "deviceCapacity": 250,
    "deviceId": "872292",
    "deviceName": "RickandMorty-MBP",
    "displayName": "JupiterOne Macbook Air",
    "enrollmentChannels": Array [
      "device",
      "user",
    ],
    "ethernetMacs": Array [
      "14:7d:da:4f:73:10",
    ],
    "id": "872292",
    "lastSeenAt": 1655845792000,
    "lastSeenIp": "[REDACTED]",
    "lastSeenOn": 1655845792000,
    "macAddress": Array [
      "14:7d:da:4f:73:10",
      "14:7D:DA:4E:50:6D",
      "14:7d:da:4f:73:10",
    ],
    "make": "Apple Inc.",
    "model": "MacBookAir9,1",
    "modelName": "MacBook Air (Retina, 13-inch, 2020)",
    "name": "JupiterOne Macbook Air",
    "osVersion": "12.0.1",
    "processorArchitecture": "Intel",
    "productName": "MacBookAir9,1",
    "serial": "C02D91CNPV4M",
    "serialNumber": "C02D91CNPV4M",
    "status": "assigned",
    "uniqueIdentifier": "644D7162-2B41-5A68-9AA7-8DB4FC6ABFF5",
    "wifiMac": "14:7d:da:4f:73:10",
  },
]
`;
